internal/kibana/role.go

```diff
@@ -0,0 +1,506 @@
+					"cluster": {
+						Description: "List of the cluster privileges.",
+						Type:        schema.TypeSet,
+						Optional:    true,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+					},
+					"indices": {
+						Description: "A list of indices permissions entries.",
+						Type:        schema.TypeSet,
+						Optional:    true,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"field_security": {
+									Description: "The document fields that the owners of the role have read access to.",
+									Type:        schema.TypeList,
+									Optional:    true,
+									MaxItems:    1,
+									Elem: &schema.Resource{
+										Schema: map[string]*schema.Schema{
+											"grant": {
+												Description: "List of the fields to grant the access to.",
+												Type:        schema.TypeSet,
+												Optional:    true,
+												Elem: &schema.Schema{
+													Type: schema.TypeString,
+												},
+											},
+											"except": {
+												Description: "List of the fields to which the grants will not be applied.",
+												Type:        schema.TypeSet,
+												Optional:    true,
+												Elem: &schema.Schema{
+													Type: schema.TypeString,
+												},
+											},
+										},
+									},
+								},
+								"query": {
+									Description:      "A search query that defines the documents the owners of the role have read access to.",
+									Type:             schema.TypeString,
+									ValidateFunc:     validation.StringIsJSON,
+									DiffSuppressFunc: utils.DiffJsonSuppress,
+									Optional:         true,
+								},
+								"names": {
+									Description: "A list of indices (or index name patterns) to which the permissions in this entry apply.",
+									Type:        schema.TypeSet,
+									Required:    true,
+									Elem: &schema.Schema{
+										Type: schema.TypeString,
+									},
+								},
+								"privileges": {
+									Description: "The index level privileges that the owners of the role have on the specified indices.",
+									Type:        schema.TypeSet,
+									Required:    true,
+									Elem: &schema.Schema{
+										Type: schema.TypeString,
+									},
+								},
+							},
+						},
+					},
+					"run_as": {
+						Description: "A list of usernames the owners of this role can impersonate.",
+						Type:        schema.TypeSet,
+						Optional:    true,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+					},
+				},
+			},
+		},
+		"kibana": {
+			Description: "The list of objects that specify the Kibana privileges for the role.",
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"base": {
+						Description: "A base privilege. When specified, the base must be [\"all\"] or [\"read\"].",
+						Type:        schema.TypeSet,
+						Optional:    true,
+						MaxItems:    1,
```

Comment:
We should be able to do something like 

```suggestion
						MaxItems:    1,
						ExactlyOneOf: []string{"kibana.base", "kibana.feature"},
```

Reply:
This is not possible due to plugin restrictions,  
[https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema#Schema](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema#Schema)  
As we use TypeSet for `"kibana"` and `"kibana.feature"` , based on the readme above, it's not possible.

  
Related:  
hashicorp/terraform-plugin-sdk#1001

hashicorp/terraform-plugin-sdk#71